lms/tasks/event.py

```diff
@@ -11,3 +23,28 @@ def insert_event(event: dict) -> None:
             request.find_service(EventService).insert_event(
                 BaseEvent(request=request, **event)
             )
+
+
+@app.task
+def purge_launch_data(*, max_age_days=30) -> None:
+    with app.request_context() as request:
+        with request.tm:
+            events_with_old_lti_params = (
+                select(Event.id)
+                .join(EventData)
+                .where(
+                    # Find data that's is at least max_age_days old
+                    Event.timestamp <= datetime.now(UTC) - timedelta(days=max_age_days),
+                    # Limit the search for only twice as old as we'd expect, limiting the data set significally
+                    Event.timestamp
+                    >= datetime.now(UTC) - timedelta(days=max_age_days * 2),
```

Comment:
This condition limits the search to only rows that are 2xmax_age_days, it's an optimization to avoid looking at the whole table all the time.

If this task stopped running we'd have a bunch of rows that are not deleted but  I reckon is a small risk.